Plugin crash / 'Plugin did not respond' errors for the 'azurerm_container_registry_task_schedule_run_now' resource

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.6.2

### AzureRM Provider Version

3.77.0

### Affected Resource(s)/Data Source(s)

azurerm_container_registry_task_schedule_run_now

### Terraform Configuration Files

```hcl
terraform {
  required_version = "1.6.2"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "3.77.0"
    }
  }
}

provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-rg"
  location = "North Europe"
}

resource "azurerm_container_registry" "example" {
  name                = "example"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  sku                           = "Standard"
  public_network_access_enabled = true
  anonymous_pull_enabled        = true
  admin_enabled                 = true

  identity {
    type = "SystemAssigned"
  }
}

resource "azurerm_container_registry_task" "purge_manifests_older_than_two_days" {
  name                  = "purge-manifests-older-than-two-days"
  container_registry_id = azurerm_container_registry.example.id

  enabled        = true
  is_system_task = false

  encoded_step {
    context_access_token = null
    context_path         = null
    task_content         = <<EOT
version: v1.1.0
steps:
  - cmd: mcr.microsoft.com/acr/acr-cli:0.8 purge --filter 'example-repository/example-image:.*' --ago 2d --untagged 
    disableWorkingDirectoryOverride: true
    timeout: 3600
EOT
  }

  platform {
    os           = "Linux"
    architecture = "amd64"
  }

  identity {
    type = "SystemAssigned"
  }

  timer_trigger {
    name     = "daily-schedule"
    schedule = "0 6 * * *" # Every day at 06:00
    enabled  = true
  }
}

resource "azurerm_role_assignment" "purge_images_acr_task_acrdelete" {
  description          = "Grant the managed identity of the '${azurerm_container_registry_task.purge_manifests_older_than_two_days.name}' ACR task permission to delete images from the Azure Container Registry (ACR)."
  scope                = azurerm_container_registry.example.id
  role_definition_name = "AcrDelete"
  principal_id         = azurerm_container_registry_task.purge_manifests_older_than_two_days.identity[0].principal_id
}

resource "azurerm_container_registry_task_schedule_run_now" "test_image_purging" {
  container_registry_task_id = azurerm_container_registry_task.purge_manifests_older_than_two_days.id
}
```


### Debug Output/Panic Output

```shell
azurerm_container_registry_task_schedule_run_now.test_image_purging: Creating...
╷
│ Error: Request cancelled
│ 
│   with azurerm_container_registry_task_schedule_run_now.test_image_purging,
│   on main.tf line 198, in resource "azurerm_container_registry_task_schedule_run_now" "test_image_purging":
│  198: resource "azurerm_container_registry_task_schedule_run_now" "test_image_purging" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵

Stack trace from the terraform-provider-azurerm_v3.77.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6aa5d5b]

goroutine 120 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/containers.(*ContainerRegistryTaskScheduleResource).Create.ContainerRegistryTaskScheduleResource.Create.func1({0x9968ed8, 0xc000208f50}, {0xc001451680, {0x996b558, 0xc000292018}, 0xc0024ff980, 0x0, {0x996b718, 0xf5618e0}})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/containers/container_registry_task_schedule_run_now_resource.go:107 +0x71b
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func2({0x9968ed8, 0xc000208f50}, 0x0?, {0x8215d80?, 0xc001451680?})
        github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:55 +0x14f
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.(*ResourceWrapper).diagnosticsWrapper.diagnosticsWrapper.func10({0x9968ed8?, 0xc000208f50?}, 0x0?, {0x8215d80?, 0xc001451680?})
        github.com/hashicorp/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:190 +0x59
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0008b8fc0, {0x9968e30, 0xc00252c450}, 0xd?, {0x8215d80, 0xc001451680})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:778 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0008b8fc0, {0x9968e30, 0xc00252c450}, 0xc0000dc9c0, 0xc0024ff800, {0x8215d80, 0xc001451680})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:909 +0xa7e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000525140, {0x9968e30?, 0xc00252c360?}, 0xc001001e50)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0002bb040, {0x9968e30?, 0xc000833950?}, 0xc0002085b0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x56a
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x8cecde0?, 0xc0002bb040}, {0x9968e30, 0xc000833950}, 0xc000208540, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002ae1e0, {0x997a560, 0xc00144c000}, 0xc0025217a0, 0xc001ad7cb0, 0xf5219f8, 0x0)
        google.golang.org/grpc@v1.58.0/server.go:1376 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc0002ae1e0, {0x997a560, 0xc00144c000}, 0xc0025217a0, 0x0)
        google.golang.org/grpc@v1.58.0/server.go:1753 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.58.0/server.go:998 +0x8d
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 23
        google.golang.org/grpc@v1.58.0/server.go:996 +0x165

Error: The terraform-provider-azurerm_v3.77.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

With azurerm v3.51.0, the resource is created successfully:

azurerm_container_registry_task_schedule_run_now.test_image_purging: Creating...
azurerm_container_registry_task_schedule_run_now.test_image_purging: Still creating... [10s elapsed]
azurerm_container_registry_task_schedule_run_now.test_image_purging: Creation complete after 12s [id=/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/example-rg/providers/Microsoft.ContainerRegistry/registries/example/tasks/purge-manifests-older-than-two-days/schedule/schedule]

### Actual Behaviour

With azurerm v3.52.0 (and azurerm v3.77.0), the request is cancelled (the plugin crashes). Please see the 'Debug Output/Panic Output' section.

### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### Important Factoids

_No response_

### References

#23334

--- a/internal/services/containers/container_registry_task_schedule_run_now_resource.go
+++ b/internal/services/containers/container_registry_task_schedule_run_now_resource.go
@@ -8,6 +8,7 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2019-06-01-preview/registries"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2019-06-01-preview/runs"
 	"github.com/hashicorp/go-azure-sdk/resource-manager/containerregistry/2019-06-01-preview/tasks"
@@ -104,8 +105,9 @@ func (r ContainerRegistryTaskScheduleResource) Create() sdk.ResourceFunc {
 
 			runName := ""
 			for _, v := range *run.Model {
-				if *v.Properties.Task == taskId.TaskName {
-					runName = *v.Name
+				if v.Properties != nil && pointer.From(v.Properties.Task) == taskId.TaskName {
+					runName = pointer.From(v.Name)
+					break
 				}
 			}
 

